### PR TITLE
Protractorenvconfig

### PR DESCRIPTION
**Pull request by traps12**
This branch includes some code refactoring for configuration files

### CODE REVIEW

File: config/config_exec/browserstack_conf/local.conf.js [hunk 1]
```diff
@@ -1,18 +1,12 @@
...
-});
+credential = require("../credentials.js")
 
 exports.config = {
    specs: ["../spec/TodoSpec.js"],
-   //seleniumAddress: 'http://hub-cloud.browserstack.com/wd/hub',
-  // webDriverProxy:'http://internetpu:8085',
    jasmineNodeOpts: {
         showColors: true, // Use colors in the command line report.
         defaultTimeoutInterval: 1000000,
```
Comment (VishalSwarnkar) [inline config/config_exec/browserstack_conf/local.conf.js]: you can have this also in the global config

File: config/config_exec/browserstack_conf/local.conf.js [hunk 2]
```diff
@@ -21,18 +15,17 @@ exports.config = {
...
+      'browserName': 'firefox',
       },
       
- // Code to start browserstack local before start of test
- beforeLaunch: function(){
+beforeLaunch: function(){
   console.log("Connecting local");
   return new Promise(function(resolve, reject){
-    bs_local.start({'key': 'FotasUZMqVHy848ZhqxE','proxy-host': 'internetpu', 'proxy-port': '8085' }, function(error) {
+    bs_local.start({'key': exports.config.capabilities['browserstack.key'],'proxy-host': credential.proxy.host, 'proxy-port': credential.proxy.port }, function(error) {
```
Comment (VishalSwarnkar) [inline config/config_exec/browserstack_conf/local.conf.js]: We don't necessarily need this proxy when we run in Jenkins or any other platform, so think of a solution where we get this on demand

File: config/config_exec/browserstack_conf/parallel.conf.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+process.env["NODE_CONFIG_DIR"] ="./config/config_env";
+let config = require('config');
 var url = config.get("web.url");
 var globalTunnel = require('global-tunnel-ng');
-globalTunnel.initialize({
-  host: 'internetpu',
-  port: 8085,
+credential = require("../credentials.js")
 
+globalTunnel.initialize({
```
Comment (VishalSwarnkar) [inline config/config_exec/browserstack_conf/parallel.conf.js:8]: look for the option where we don't necessarily need every time of execution

File: config/config_exec/common.conf.js [hunk 4]
```diff
@@ -1,24 +1,24 @@
 var Jasmine2HtmlReporter = require("protractor-jasmine2-html-reporter");
-const config = require('config'); 
+process.env["NODE_CONFIG_DIR"] ="./config/config_env";
+let config = require('config'); 
 var url = config.get("web.url")
 
-exports.config = {
-    specs: ["../spec/TodoSpec.js"],
+module.exports = {
```
Comment (VishalSwarnkar) [inline config/config_exec/common.conf.js:6]: good

File: config/config_exec/credentials.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+"browserstack":{
+        "username":"vishal538",
+        "key":"NqQyPLgqydEAegmxE5uh",
```
Comment (VishalSwarnkar) [inline config/config_exec/credentials.js]: please remove the access key

File: helper/CalculationHelper.js [hunk 6]
```diff
@@ -5,20 +5,7 @@ var myProfile = new FirefoxProfile();
...
-   //          capabilities.set("firefox_profile", encodedProfile);
-   //  var wd = new webdriver.Builder().
-   //                    withCapabilities(capabilities).
-   //                    build();
-            
-   //  wd.get("http://juliemr.github.io/protractor-demo/");  });
-    };
-
-   this.clickButton = function(locator) {
+  this.clickButton = function(locator) {
```
Comment (VishalSwarnkar) [inline helper/CalculationHelper.js:8]: good cleanup

File: package.json [hunk 7]
```diff
@@ -17,12 +17,11 @@
     "config": "^3.0.0"
   },
   "scripts": {
-    "test": "set NODE_ENV=test&node node_modules/protractor/bin/protractor conf/docker-conf.js",
-    "production": "set NODE_ENV=production&node node_modules/protractor/bin/protractor conf.js",
-    "localtest": "set NODE_ENV=production&node node_modules/protractor/bin/protractor conf/local.conf.js",
-    "singletest": "set NODE_ENV=production&node node_modules/protractor/bin/protractor conf/single.conf.js",
-    "paralleltest": "set NODE_ENV=production&node node_modules/protractor/bin/protractor conf/parallel.conf.js",
-    "docker-test": "set NODE_ENV=production&node node_modules/protractor/bin/protractor conf/docker-conf.js"
+    "test": "set NODE_ENV=test&node node_modules/protractor/bin/protractor config/config_exec/protractor_conf/simple.conf.js",
```
Comment (VishalSwarnkar) [inline package.json:20]: We have too many options, this is temporary once we have actual implementation we may don't need this many.

File: spec/TodoSpec.js [hunk 8]
```diff
@@ -5,10 +5,12 @@ let number2 = 5;
 
 var calculationPage = new CalculationPage();
 var calculationHelper = new CalculationHelper();
+beforeEach(function() {
```
Comment (VishalSwarnkar) [inline spec/TodoSpec.js:8]: With the current application, this is ideal but with CRM application we may need a function which will construct the URL if needed with multiple parameters

Comment (VishalSwarnkar) [inline spec/TodoSpec.js:8]: Please remove all the screenshots

File: config/config_exec/browserstack_conf/local.conf.js (hunk 1 above)
Comment (traps12) [reply config/config_exec/browserstack_conf/local.conf.js]: Done, Added timeout variable in credentials.js , which is common env variable file

File: config/config_exec/browserstack_conf/local.conf.js (hunk 2 above)
Comment (traps12) [reply config/config_exec/browserstack_conf/local.conf.js]: Done, now it is on demand

File: config/config_exec/browserstack_conf/parallel.conf.js (hunk 3 above)
Comment (traps12) [reply config/config_exec/browserstack_conf/parallel.conf.js:8]: It is only require for browser stack cloud

File: config/config_exec/credentials.js (hunk 5 above)
Comment (traps12) [reply config/config_exec/credentials.js]: Removed it, Need to discuss more about this , as you suggested provide this  by environment variable but same issue with windows and linux command line

File: spec/TodoSpec.js (hunk 8 above)
Comment (traps12) [reply spec/TodoSpec.js:8]: Done

**Pull request merged**